pkg/agent/nodeportlocal/rules/iptable_rule_test.go

```diff
@@ -0,0 +1,163 @@
+				mockIPTables.EnsureChain(iptables.ProtocolIPv4, iptables.NATTable, NodePortLocalChain)
+				mockIPTables.AppendRule(iptables.ProtocolIPv4, iptables.NATTable, iptables.PreRoutingChain, []string{"-p", "all", "-m", "addrtype", "--dst-type", "LOCAL", "-j", NodePortLocalChain})
+				mockIPTables.AppendRule(iptables.ProtocolIPv4, iptables.NATTable, iptables.OutputChain, []string{"-p", "all", "-m", "addrtype", "--dst-type", "LOCAL", "-j", NodePortLocalChain})
+				mockIPTables.Restore(`*nat
+:ANTREA-NODE-PORT-LOCAL - [0:0]
+COMMIT
+`, false, false)
+				mockIPTables.DeleteRule(iptables.ProtocolIPv4, iptables.NATTable, iptables.PreRoutingChain, []string{"-p", "all", "-m", "addrtype", "--dst-type", "LOCAL", "-j", NodePortLocalChain})
+				mockIPTables.DeleteRule(iptables.ProtocolIPv4, iptables.NATTable, iptables.OutputChain, []string{"-p", "all", "-m", "addrtype", "--dst-type", "LOCAL", "-j", NodePortLocalChain})
+				mockIPTables.DeleteChain(iptables.ProtocolIPv4, iptables.NATTable, NodePortLocalChain)
+			},
+		},
+		{
+			name: "valid rule",
+			rules: []PodNodePort{
+				{
+					NodePort:  7,
+					PodPort:   17,
+					PodIP:     "localhost",
+					Protocol:  "tcp",
+					Protocols: []string{"tcp", "udp"},
+				},
+			},
+			expectedCalls: func(mockIPTables *iptablestest.MockInterfaceMockRecorder) {
+				mockIPTables.EnsureChain(iptables.ProtocolIPv4, iptables.NATTable, NodePortLocalChain)
+				mockIPTables.AppendRule(iptables.ProtocolIPv4, iptables.NATTable, iptables.PreRoutingChain, []string{"-p", "all", "-m", "addrtype", "--dst-type", "LOCAL", "-j", NodePortLocalChain})
+				mockIPTables.AppendRule(iptables.ProtocolIPv4, iptables.NATTable, iptables.OutputChain, []string{"-p", "all", "-m", "addrtype", "--dst-type", "LOCAL", "-j", NodePortLocalChain})
+				mockIPTables.Restore(`*nat
+:ANTREA-NODE-PORT-LOCAL - [0:0]
+-A ANTREA-NODE-PORT-LOCAL -p tcp -m tcp --dport 7 -j DNAT --to-destination localhost:17
+-A ANTREA-NODE-PORT-LOCAL -p udp -m udp --dport 7 -j DNAT --to-destination localhost:17
+COMMIT
+`, false, false)
+				mockIPTables.DeleteRule(iptables.ProtocolIPv4, iptables.NATTable, iptables.PreRoutingChain, []string{"-p", "all", "-m", "addrtype", "--dst-type", "LOCAL", "-j", NodePortLocalChain})
+				mockIPTables.DeleteRule(iptables.ProtocolIPv4, iptables.NATTable, iptables.OutputChain, []string{"-p", "all", "-m", "addrtype", "--dst-type", "LOCAL", "-j", NodePortLocalChain})
+				mockIPTables.DeleteChain(iptables.ProtocolIPv4, iptables.NATTable, NodePortLocalChain)
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			mockIPTables := iptablestest.NewMockInterface(ctrl)
+			rules := iptablesRules{
+				name:  "test-rules",
+				table: mockIPTables,
+			}
+
+			gomock.InOrder(
+				// Set the expectation for ChainExists to return true
+				mockIPTables.EXPECT().ChainExists(iptables.ProtocolIPv4, iptables.NATTable, NodePortLocalChain).Return(true, nil),
```

Comment:
It's better to move this expectation to `expectedCalls` as well to consolidate expectations.